docs/contributing.rst

```diff
@@ -253,6 +253,310 @@ Commit messages should fit the following template.
+say that in your comment so that the PR author
+doesn't make that change in the meantime.
+
+A bad use of an inline comment is to ask for
+a major change to the PR.
+These comments tend to be
+frustrating for PR authors to respond to,
+and in general are unbalanced in terms of reviewer effort
+versus PR author effort.
+For major changes, instead make a FIXUP commit.
+
+Inline comments should not block the merging of a PR.
+The maintainer merging the PR will make the typo / style fixes
+they deem appropriate during the merge
+if the PR author doesn't get around to fixing them.
+If the discussion raises a new issue or feature request,
+make a new issue to track that so that it doesn't
+block PR progress.
+
+Making commits
+--------------
+
+Instead of asking the PR author for changes,
+we prefer reviewers make commits
+to propose changes to a PR.
+Commits allow the reviewer to propose explicit
+changes that the PR author can say yes or no to,
+rather than placing the burden on the PR author
+and allowing for miscommunication.
+
+In most cases, PRs are made from a feature branch
+to master in the same repository,
+in which case you can push commits
+to the feature branch directly.
+If the PR comes from a fork,
+you may have to make a PR
+on the feature branch in their repo.
+
+There are four types of commits that reviewers
+can add, depending on the type of change proposed.
+
+1. ``fixup`` commits should be used for minor changes
+   like style fixes, moving code from one location to another,
+   and fixing small bugs.
+   In the end, your ``fixup`` commit will not appear in
+   the history of the ``master`` branch.
+
+   To make a ``fixup`` commit, first make the desired changes
+   and ``git add`` them. When making the commit, do
+   ``git commit --fixup <commit hash>`` where the commit hash
+   corresponds to the commit that your ``fixup`` commit modifies.
+
+   ``fixup`` commits are so named because
+   the maintainer will ``fixup`` those commits into
+   the appropriate part of the PR branch's history
+   before merging that branch into ``master``.
+
+2. ``squash`` commits should be used for minor changes
+   that require some explanation.
+   In the end, your ``squash`` commit will not appear in
+   the history of the ``master`` branch,
+   except in one or more commit messages.
+
+   To make a ``squash`` commit, first make the desired changes
+   and ``git add`` them. When making the commit, do
+   ``git commit --squash <commit hash>`` where the commit hash
+   corresponds to the commit that your ``squash`` commit modifies.
+   Unlike with the ``--fixup`` option, git will now prompt you
+   to enter a message to explain what your ``squash`` commit does.
+
+   ``squash`` commits are so named because
+   the maintainer will ``squash`` those commits into
+   the appropriate part of the PR branch's history
+   before merging that branch into ``master``.
+   Since ``squash`` commits contain a commit message,
```

Comment:
Just noticed this when getting ready to merge, looks like the end of this sentence disappeared somewhere